src/policy/sft_map.rs

```diff
@@ -348,49 +348,51 @@ mod dense_chunk_map {
         }
     }
 
-    impl<'a> SFTDenseChunkMap<'a> {
+    impl SFTDenseChunkMap {
         /// Empty space is at index 0
         const EMPTY_SFT_INDEX: u8 = 0;
 
         pub fn new() -> Self {
             Self {
                 /// Empty space is at index 0
-                sft: vec![&EMPTY_SPACE_SFT],
-                index_map: HashMap::new(),
+                sft: UnsafeCell::new(vec![&EMPTY_SPACE_SFT]),
+                index_map: UnsafeCell::new(HashMap::new()),
             }
         }
 
-        // This is a temporary solution to allow unsafe mut reference. We do not want several occurrence
-        // of the same unsafe code.
-        // FIXME: We need a safe implementation.
-        #[allow(clippy::cast_ref_to_mut)]
-        #[allow(clippy::mut_from_ref)]
-        unsafe fn mut_self(&self) -> &mut Self {
-            &mut *(self as *const _ as *mut _)
-        }
-
         pub fn addr_to_index(addr: Address) -> u8 {
             SFT_DENSE_CHUNK_MAP_INDEX.load_atomic::<u8>(addr, Ordering::Relaxed)
         }
+        #[inline(always)]
```

Comment:
Could you remove the `#[inline(always)]` here and for `index_map`? We've discovered that adding inline pragmas isn't always ideal and we prefer using profile-guided optimizations nowadays.

Reply:
Done

Reply:
Thanks :)